### ISSUE TITLE
Click-dragging in between multiple-selected elements should not deselect

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#1898]
Click-dragging in between elements, but still within the common bounding box, should drag the whole group, not deselect. This is the standard behavior (Figma, Affinity,...).

Desired behavior:

![excal_group_drag](https://user-images.githubusercontent.com/5153846/87017368-2bc65d00-c1d0-11ea-8da2-f044eabfadd4.gif)

In fact, this concerns even single-element bounding boxes, such as dragging a diamond or custom shape by its bounding box (outside the shape).

Figma:

![excal_group_drag_triangle](https://user-images.githubusercontent.com/5153846/87017781-b73fee00-c1d0-11ea-9faa-26b10cc6ed15.gif)

**Comment by Jnforja** [comment]
Hi!
I'd like to work on this issue, has it been taken yet?

@dwelle

**Comment by dwelle** [comment]
Go ahead. Make sure the hitbox aligns with the bounding box only after the element(s) are already selected.

**Comment by Jnforja** [comment]
> Go ahead. Make sure the hitbox aligns with the bounding box only after the element(s) are already selected.

Ok! Thanks for calling my attention to that.

**Comment by Jnforja** [comment]
After looking at the code, I concluded that it might be better to do this with multiple PRs, so they're easier to review and easier to merge.

If I'm not mistaken, to fix the "single-element bounding box bug", I'll have to change `getElementAtPosition()` at file comparisons.ts. Since `getElementAtPosition()` isn't covered by tests, I'd like to write some to avoid introducing regressions. So I'd do a PR to add the tests, and another to fix this bug.

Regarding the bug "click-dragging in between elements", I'm still not 100% sure how to solve it without introducing some strange edge-cases. But I think I'll have a better idea after fixing the "single-element bounding box bug".

In summary, I'd do the following:
1. PR to add automated tests for  `getElementAtPosition()`.
2. PR to fix single-element bounding box bug
3. Come up with an approach to fix the "click-dragging in between elements" bug.

@dwelle is this ok with you?

**Comment by dwelle** [comment]
Sounds good.

**Comment by Jnforja** [comment]
I think we can close this issue :)

**Comment by dwelle** [comment]
Indeed!
